src/string.rs

```diff
@@ -0,0 +1,398 @@
+#[derive(Debug, Clone, Copy)]
+pub struct StringCursor {
+    pub(crate) ptr: *const u8,
+    pub position: usize,
+}
```

Comment:
Is this `&[u8]` or different to it?

e.g. will this version work?
```
pub struct StringCursor {
    bytes: &[u8],
}

cursor.bytes = ...
```

Reply:
Nop, position is character position, ptr is bytes cursor.